cassiopeia/ProcessingPipeline/process/pipeline.py

```diff
@@ -98,33 +100,150 @@ def resolve_UMI_sequence(
 
     # apply the filter using the hash table created above
     molecule_table["filter"] = molecule_table["readName"].map(mt_filter)
-    n_filtered = molecule_table[molecule_table['filter'] == True].shape[0]
+    n_filtered = molecule_table[molecule_table["filter"] == True].shape[0]
 
-    logging.info(f'Filtered out {n_filtered} reads.')
+    logging.info(f"Filtered out {n_filtered} reads.")
 
     # filter based on status & reindex
-    filt_molecule_table = molecule_table[
-        molecule_table["filter"] == False
-    ].copy()
+    filt_molecule_table = molecule_table[molecule_table["filter"] == False].copy()
     filt_molecule_table.drop(columns=["filter"], inplace=True)
 
-    logging.info(f'Finished resolving UMI sequences in {time.time() - t0}s.')
+    logging.info(f"Finished resolving UMI sequences in {time.time() - t0}s.")
 
     # ---------------- Plot Diagnositics after Resolving ---------------- #
     h = plt.figure(figsize=(14, 10))
     plt.plot(top_reads.values(), total_numReads.values(), "r.")
     plt.ylabel("Total Reads")
     plt.xlabel("Number Reads for Picked Sequence")
     plt.title("Total vs. Top Reads for Picked Sequence")
-    plt.savefig(os.path.join(outputdir, "/total_vs_top_reads_pickSeq.png"))
+    plt.savefig(os.path.join(output_directory, "/total_vs_top_reads_pickSeq.png"))
     plt.close()
 
     h = plt.figure(figsize=(14, 10))
     plt.plot(first_reads.values(), second_reads.values(), "r.")
     plt.ylabel("Number Reads for Second Best Sequence")
     plt.xlabel("Number Reads for Picked Sequence")
     plt.title("Second Best vs. Top Reads for Picked Sequence")
-    plt.savefig(os.path.join(outputdir + "/second_vs_top_reads_pickSeq.png"))
+    plt.savefig(os.path.join(output_directory + "/second_vs_top_reads_pickSeq.png"))
     plt.close()
 
     return filt_molecule_table
+
+
+CELL_BC_TAG = "CB"
```

Comment:
Perhaps pass these to the function `collapseUMIs`? I'm not sure we should define these as global variables.